### PR TITLE
issue role certificate with Athenz server default expiry for non-CopperArgos cases

### PR DESCRIPTION
**Pull request by WindzCUHK**
# Description

For availability, the role certificate expiry should be reasonably long. For non-CopperArgos cases, when the external certificate is short-lived (< 24h), the role certificate issued may expire before the default refreshing cycle (24h). 

## Assignees
- [x] `Assignees` is set

## Type of changes
- [x] Apply one or more `labels` of the following that fits:
  - `bug`: Bug fix
  - `dependencies`: Dependency upgrades
  - `documentation`: Documentation changes
  - `enhancement`: New Feature
  - `good first issue`: First contribution
  - `logging`: Log changes
  - `refactor`: Refactoring (no functional changes, no api changes)

### Flags
```
- [ ] Breaks backward compatibility
- [ ] Requires a documentation update
- [ ] Has untestable code
```
---

## Checklist
```
- [ ] Followed the guidelines in the CONTRIBUTING document
- [ ] Added prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]` in the PR title if necessary
- [ ] Tested and linted the code
- [ ] Commented the code
- [ ] Made corresponding changes to the documentation
```

## Checklist for maintainer
```
- [ ] Use `Squash and merge`
- [ ] Double-confirm the merge message has prefix `[skip ci]`/`[ci skip]`/`[no ci]`/`[skip actions]`/`[actions skip]`
- [ ] Delete the branch after merge
```

### CODE REVIEW

File: pkg/certificate/identity.go [hunk 1]
```diff
@@ -287,9 +288,14 @@ func (h *identityHandler) GetX509RoleCert() (rolecerts [](*RoleCertificate), rol
...
 		}
+
+		var roleCertExpiry int64 // 0 implies using server default expiry time
+		if h.idCfg.ServiceCert.CopperArgos.Use {
+			roleCertExpiry = int64(x509LeafCert.NotAfter.Sub(time.Now()).Minutes()) + int64(config.DEFAULT_ROLE_CERT_EXPIRY_TIME_BUFFER_MINUTES) // Extract NotAfter from the instance certificate
+		}
 		roleRequest := &zts.RoleCertificateRequest{
 			Csr:        string(roleCsrPEM),
-			ExpiryTime: int64(x509LeafCert.NotAfter.Sub(time.Now()).Minutes()) + int64(config.DEFAULT_ROLE_CERT_EXPIRY_TIME_BUFFER_MINUTES), // Extract NotAfter from the instance certificate
+			ExpiryTime: roleCertExpiry,
```
Comment (mlajkim) [inline pkg/certificate/identity.go:298]: (Optional) Consider using anonymous function to make it more readable
```
ExpiryTime: func() {
  return ...
}(),
```

Comment (WindzCUHK) [reply pkg/certificate/identity.go]: fixed in [fa2aae3](https://github.com/AthenZ/k8s-athenz-sia/pull/178/commits/fa2aae34bae2492317f3fa61d96315f58ff745f7)

File: pkg/certificate/identity.go [hunk 2]
```diff
@@ -287,9 +288,14 @@ func (h *identityHandler) GetX509RoleCert() (rolecerts [](*RoleCertificate), rol
 		if err != nil {
 			return nil, nil, fmt.Errorf("Failed to prepare csr, failed to parse certificate for PostRoleCertificateRequest, Subject CommonName[%s], err: %v", csrOption.Subject.CommonName, err)
 		}
+
+		var roleCertExpiry int64 // 0 implies using server default expiry time
```
[the remark this answers is not in the document]
Comment (WindzCUHK) [reply pkg/certificate/identity.go]: I think the original case is special and need explanation instead. Added comment for that in [fa2aae3](https://github.com/AthenZ/k8s-athenz-sia/pull/178/commits/fa2aae34bae2492317f3fa61d96315f58ff745f7)

[the remark this answers is not in the document]
Comment (mlajkim) [reply pkg/certificate/identity.go]: @WindzCUHK sounds good. lgtm

**Review by mlajkim — APPROVED** [review]
lgtm

**Pull request merged**